### PR TITLE
[Substrate.io Migration] Tutorials -> Build a Blockchain -> Simulate a Network

### PR DESCRIPTION
**Pull request by 0xLucca**
The "Simulate a Network" page was migrated from the existing Substrate.io Tutorial section: https://docs.substrate.io/tutorials/build-a-blockchain/simulate-network/

### CODE REVIEW

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+description: Create a private blockchain network with authorized validators using Polkadot-SDK. Start nodes, connect peers and verify block production.
+---
+
+# Simulate a Network
+
+## Introduction
+
+This tutorial introduces you to the process of initiating a private blockchain network with a set of authorized validators. The [Polkadot-SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank} implements an authority consensus model to regulate block production.
+
+In this model, the creation of blocks is restricted to a predefined list of authorized accounts, known as "authorities," who operate in a round-robin fashion.
```
Comment (nhussein11) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:12]:
```suggestion
In this model, the creation of blocks is restricted to a predefined list of authorized accounts, known as "authorities", who operate in a round-robin fashion. 
```

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+- Start a blockchain node using a predefined account
+- Learn the key command-line options used to start a node
+- Determine if a node is running and producing blocks
+- Connect a second node to a running network
+- Verify peer computers produce and finalize blocks
+
+## Prerequisites
+
+- Installed and configured Rust on your system. For detailed instructions on installing Rust and setting up your development environment, refer to the [Installation]() guide.
+- Completed the [Build a Local Blockchain](#build-a-local-blockchain) guide and have the [Polkadot-SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank} installed on your local machine.
```
Comment (nhussein11) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:27]:
```suggestion
- Installed and configured Rust on your system. For detailed instructions on installing Rust and setting up your development environment, refer to the [Installation]() guide
- Completed the [Build a Local Blockchain](#build-a-local-blockchain) guide and have the [Polkadot-SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank} installed on your local machine
```

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+- Start a blockchain node using a predefined account
+- Learn the key command-line options used to start a node
+- Determine if a node is running and producing blocks
+- Connect a second node to a running network
+- Verify peer computers produce and finalize blocks
+
+## Prerequisites
+
+- Installed and configured Rust on your system. For detailed instructions on installing Rust and setting up your development environment, refer to the [Installation]() guide.
```
Comment (nhussein11) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:26]:
```suggestion
- Installed and configured Rust on your system. Refer to the [Installation]() guide for detailed instructions on installing Rust and setting up your development environment.
```
I think this is clearer, however feel free to decide based on your own taste for this one

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+## Start the First Blockchain Node
+
+This tutorial demonstrates the fundamentals of a private network using a predefined chain specification called local and two preconfigured user accounts. You'll simulate a private network by running two nodes on a single local computer, using accounts named Alice and Bob.
+
+Follow these steps to start your first blockchain node:
+
+1. Open a terminal on your computer
+   
+2. Navigate to the root directory where you compiled the Polkadot-SDK Solochain Template
```
Comment (nhussein11) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:37]: I'd merge 1. and 2. into one single step

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+```bash
+./target/release/solochain-template-node --help
+```
+
+## Review the Node Messages
+
+Upon successful node startup, the terminal displays messages detailing network operations and information relevant to the running node. This output includes details about the chain specification, system data, network status, and other crucial parameters. You should see output similar to this:
+
+<div id='termynal' data-termynal>
+    <span data-ty>2024-09-10 08:35:43 Substrate Node</span>
```
Comment (nhussein11) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:81]: The command that produces this output is missing in this `termynal` element

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+
+    This message indicates that:
+
+    - No other nodes are currently in the network
+    - No blocks are being produced
+    - Block production will commence once another node joins the network
+
+## Add a Second Node to the Network
+
+After successfully running the first node with the alice account keys, you can expand the network by adding a second node using the bob account. This process involves connecting to the existing network using the running node as a reference point. The commands are similar to those used for the first node, with some key differences to ensure proper network integration.
```
Comment (nhussein11) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:133]: You previously referred to Alice and Bob using uppercase; now, you are using lowercase. I'd stick to only one format.

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+## Verify Blocks are Produced and Finalized
+
+After starting the second node, both nodes should connect as peers and commence block production.
+
+Follow these steps to verify that blocks are being produced and finalized:
+
+1. Observe the output in the terminal of the first node (Alice):
+
+    <div id='termynal' data-termynal>
+        <span data-ty>2024-09-10 09:04:57 discovered: 12D3KooWHdiAxVd8uMQR1hGWXccidmfCwLqcMpGwR6QcTP6QRMuD /ip4/192.168.1.4/tcp/30334</span>
```
Comment (nhussein11) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:188]: This is super nitpicking, i'd add a line with "..." if this command has some lines above that you just skipped

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+3. Shut down one node using `Control-C` in its terminal. Observe the remaining node's output:
+
+    <div id='termynal' data-termynal>
+        <span data-ty>2024-09-10 09:10:03 💤 Idle (1 peers), best: #51 (0x0dd6…e763), finalized #49 (0xb70a…1fc0), ⬇ 0.7kiB/s ⬆ 0.6kiB/s</span>
+        <span data-ty>2024-09-10 09:10:08 💤 Idle (0 peers), best: #52 (0x2c40…a50e), finalized #49 (0xb70a…1fc0), ⬇ 0.3kiB/s ⬆ 0.3kiB/s</span>
+    </div>
+
+    Note that the peer count drops to zero and block production stops.
+
+4. Shut down the second node using `Control-C` in its terminal.
```
Comment (nhussein11) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:224]:
```suggestion
4. Shut down the second node using `Control-C` in its terminal
```

**Comment by 0xLucca** [comment]
We could add one more section that runs all that was done in this guide with Zombienet. We could just link to the Zombienet page, show the commands that we execute, and that the network is up and running.

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+description: Create a private blockchain network with authorized validators using Polkadot-SDK. Start nodes, connect peers and verify block production.
+---
+
+# Simulate a Network
+
+## Introduction
+
+This tutorial introduces you to the process of initiating a private blockchain network with a set of authorized validators. The [Polkadot-SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank} implements an authority consensus model to regulate block production.
+
+In this model, the creation of blocks is restricted to a predefined list of authorized accounts, known as "authorities", who operate in a round-robin fashion.
```
Comment (dawnkelly09) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:12]:
```suggestion
In this model, the creation of blocks is restricted to a predefined list of authorized accounts, known as "authorities," who operate in a round-robin fashion. 
```
This actually is true about the comma being inside the quotation marks. I know we keep it on the outside for things in backticks but, if this is in quotes the comma goes on the inside.

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+
+```bash
+./target/release/solochain-template-node --help
+```
+
+## Review the Node Messages
+
+Upon successful node startup, the terminal displays messages detailing network operations and information relevant to the running node. This output includes details about the chain specification, system data, network status, and other crucial parameters. You should see output similar to this:
+
+<div id='termynal' data-termynal>
```
Comment (dawnkelly09) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:78]: Please move the terminal outputs to snippets. Thank you!

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 11]
```diff
@@ -0,0 +1,243 @@
...
+
+## Verify Blocks are Produced and Finalized
+
+After starting the second node, both nodes should connect as peers and commence block production.
+
+Follow these steps to verify that blocks are being produced and finalized:
+
+1. Observe the output in the terminal of the first node (`Alice`):
+
+    <div id='termynal' data-termynal>
```
Comment (dawnkelly09) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:194]: Please move the terminal outputs to snippets. Thank you!

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 12]
```diff
@@ -0,0 +1,243 @@
...
+title: Simulate a Network
+description: Create a private blockchain network with authorized validators using Polkadot-SDK. Start nodes, connect peers and verify block production.
+---
+
+# Simulate a Network
+
+## Introduction
+
+This tutorial introduces you to the process of initiating a private blockchain network with a set of authorized validators. The [Polkadot-SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank} implements an authority consensus model to regulate block production.
+
```
Comment (dawnkelly09) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:11]:
```suggestion
```
It looks like this is supposed to be all one paragraph possibly? If so, we should remove the blank line.

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 13]
```diff
@@ -0,0 +1,243 @@
...
+## Start the First Blockchain Node
+
+This tutorial demonstrates the fundamentals of a private network using a predefined chain specification called local and two preconfigured user accounts. You'll simulate a private network by running two nodes on a single local computer, using accounts named `Alice` and `Bob`.
+
+Follow these steps to start your first blockchain node:
+   
+1. Navigate to the root directory where you compiled the Polkadot-SDK Solochain Template
+   
+2. Clear any existing chain data by executing:
```
Comment (dawnkelly09) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:37]:
```suggestion
2. Clear any existing chain data by executing the following:
```

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 14]
```diff
@@ -0,0 +1,243 @@
...
+2. Check the terminal of the second node (`Bob`) for similar output
+
+3. Shut down one node using `Control-C` in its terminal. Observe the remaining node's output:
+
+    <div id='termynal' data-termynal>
+        <span data-ty>2024-09-10 09:10:03 💤 Idle (1 peers), best: #51 (0x0dd6…e763), finalized #49 (0xb70a…1fc0), ⬇ 0.7kiB/s ⬆ 0.6kiB/s</span>
+        <span data-ty>2024-09-10 09:10:08 💤 Idle (0 peers), best: #52 (0x2c40…a50e), finalized #49 (0xb70a…1fc0), ⬇ 0.3kiB/s ⬆ 0.3kiB/s</span>
+    </div>
+
+    Note that the peer count drops to zero and block production stops.
```
Comment (dawnkelly09) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:230]:
```suggestion
    Note that the peer count drops to zero, and block production stops.
```

**Review by dawnkelly09 — CHANGES_REQUESTED** [review]
I have not run this for tech accuracy but, I did add some comments on formatting/Grammarly type stuff. Thanks!

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 15]
```diff
@@ -0,0 +1,243 @@
...
+### Review the Command-Line Options
+
+Before proceeding, examine the key command-line options used to start the node:
+
+- `--base-path` - specifies the directory for storing all chain-related data
+- `--chain` - defines the chain specification to use
+- `--alice` - adds the predefined keys for the `Alice` account to the node's keystore. This account is used for block production and finalization
+- `--port` - sets the listening port for peer-to-peer (p2p) traffic. Different ports are necessary when running multiple nodes on the same machine
+- `--rpc-port` - specifies the port for incoming JSON-RPC traffic via WebSocket and HTTP
+- `--node-key` - defines the Ed25519 secret key for libp2p networking
```
Comment (CrackTheCode016) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:65]: Just double checking, I think nodes no longer auto produce keys anymore, hence this command? There is also an unsafe flag that can be used to auto-gen a node key for this purpose as well (`--unsafe-force-node-key-generation`)

Comment (CrackTheCode016) [reply tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:65]: Actually, in order to have a consistent p2p address, your approach I think makes more sense.

**Review by CrackTheCode016 — APPROVED** [review]
LGTM, tested the tutorial, working great

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md (hunk 12 above)
Comment (0xLucca) [reply tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:11]: Leaving the empty line forces the text to start on a new line. I think it looks better this way

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 16]
```diff
@@ -0,0 +1,236 @@
...
+description: Create a private blockchain network with authorized validators using Polkadot-SDK. Start nodes, connect peers and verify block production.
+---
+
+# Simulate a Network
+
+## Introduction
+
+This tutorial introduces you to the process of initiating a private blockchain network with a set of authorized validators. The [Polkadot-SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank} implements an authority consensus model to regulate block production.
+
+In this model, the creation of blocks is restricted to a predefined list of authorized accounts, known as "authorities", who operate in a round-robin fashion.
```
[the remark this answers is not in the document]
Comment (0xLucca) [reply tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:12]: I don't understand this one

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md (hunk 15 above)
Comment (0xLucca) [reply tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:65]: I struggled a bit with that, but I ended up hardcoding the key to maintain the p2p address of at least the first node that works as a bootnode for the other ones

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+Follow these steps to start your first blockchain node:
+   
+1. Navigate to the root directory where you compiled the Polkadot-SDK Solochain Template
+   
+2. Clear any existing chain data by executing the following:
+    ```bash
+    ./target/release/solochain-template-node purge-chain --base-path /tmp/alice --chain local
+    ```
+
+    When prompted to confirm, type `y` and press *Enter*. This step ensures a clean start for your new network.
```
[the remark this answers is not in the document]
Comment (eshaben) [reply tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:42]: Control + C is in back ticks later in this guide, let's just go with that for now. I'll have to implement some style updates so we can render these keys with the kbd styling

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 18]
```diff
@@ -0,0 +1,197 @@
...
+    <div id='termynal' data-termynal>
+        <span data-ty>2024-09-10 09:10:03 💤 Idle (1 peers), best: #51 (0x0dd6…e763), finalized #49 (0xb70a…1fc0), ⬇ 0.7kiB/s ⬆ 0.6kiB/s</span>
+        <span data-ty>2024-09-10 09:10:08 💤 Idle (0 peers), best: #52 (0x2c40…a50e), finalized #49 (0xb70a…1fc0), ⬇ 0.3kiB/s ⬆ 0.3kiB/s</span>
+    </div>
+
+    Note that the peer count drops to zero, and block production stops.
+
+4. Shut down the second node using `Control-C` in its terminal
+
+5. Clean up chain state. To remove the chain state from the simulated network, use the `purge-chain` subcommand:
```
Comment (eshaben) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:188]:
```suggestion
5. Clean up chain state from the simulated network by using the `purge-chain` subcommand:
```

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 19]
```diff
@@ -0,0 +1,197 @@
...
+
+    - For Alice's node:
+        ```bash
+        ./target/release/solochain-template-node purge-chain --base-path /tmp/alice --chain local -y
+        ```
+    - For Bob's node:
+        ```bash
+        ./target/release/solochain-template-node purge-chain --base-path /tmp/bob --chain local -y
+        ```
```
Comment (eshaben) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:197]: Can you please break these up into several lines so you don't have to horizontally scroll to see the entire command

File: tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md [hunk 20]
```diff
@@ -0,0 +1,197 @@
...
+    - Block finalization - `finalized #1 (0x75bb…e82d)`
+
+2. Check the terminal of the second node (`Bob`) for similar output
+
+3. Shut down one node using `Control-C` in its terminal. Observe the remaining node's output:
+
+    <div id='termynal' data-termynal>
+        <span data-ty>2024-09-10 09:10:03 💤 Idle (1 peers), best: #51 (0x0dd6…e763), finalized #49 (0xb70a…1fc0), ⬇ 0.7kiB/s ⬆ 0.6kiB/s</span>
+        <span data-ty>2024-09-10 09:10:08 💤 Idle (0 peers), best: #52 (0x2c40…a50e), finalized #49 (0xb70a…1fc0), ⬇ 0.3kiB/s ⬆ 0.3kiB/s</span>
+    </div>
```
Comment (eshaben) [inline tutorials/polkadot-sdk/build-a-blockchain/simulate-a-network.md:182]: This should be a snippet

**Review by eshaben — COMMENTED** [review]
Nice work!

**Review by eshaben — APPROVED** [review]
LGTM!

**Review by dawnkelly09 — APPROVED** [review]
LGTM. Nice work, all!

**Pull request merged**